### PR TITLE
fix: stale pageSize cause of missing useEffect deps

### PR DESCRIPTION
**Pull request by astahmer**
docs: add MinimalExample

Closes : none, I didnt open an issue

## Description

adding missing useEffects deps, you might want to add the [react-hooks/exhaustive-deps eslint rule](https://www.npmjs.com/package/eslint-plugin-react-hooks) for that, I noticed this isnt the only place with missing deps

## Current Tasks

none

## Live Demo Link

i did add a [MinimalExample](https://github.com/astahmer/mantine-data-grid/commit/72997ae46961a20f32ee69cbb7d70d9c6ee497d0#diff-493e0112b69a987c1da637f97e9623f3d212b206e8e344bf63a281691e34837e) that was buggy before that PR, all the current examples use `withPagination` which solved the problem

basically it goes like this, with an initial state like:
Screenshot 2022-10-14 at 00 10 00

before the PR:
Screenshot 2022-10-14 at 00 10 06

after the PR:
Screenshot 2022-10-14 at 00 13 21

### CODE REVIEW

**Comment by Kuechlin** [comment]
looks good, i added eslint-plugin-react-hooks and fixed the hooks with missing deps

**Review by Kuechlin — APPROVED** [review]
